earthfile2llb/interpreter.go

```diff
@@ -288,6 +288,8 @@ func (i *Interpreter) handleWith(ctx context.Context, with spec.WithStatement) e
 }
 
 func (i *Interpreter) handleIf(ctx context.Context, ifStmt spec.IfStatement) error {
+	defer i.converter.markFakeDeps()
```

Comment:
Looking through the code, I don't see how this could help. This should anyway be called via `converter.FinalizeStates`, which is automatically called at the end of the interpretation of a target.